src/Core/Scheduler/Providers/Raster_Provider.js

```diff
@@ -73,81 +51,64 @@ export default {
-                    } else if (file.getElementsByTagName('gpx')[0]) {
-                        // GPX crs specification : 'EPSG:4326'
-                        layer.options.mimetype = 'vector/gpx';
-                        layer.projection = layer.projection || 'EPSG:4326';
-                    } else {
-                        throw new Error('Unsupported xml file data vector');
-                    }
-                // Know if it's an geojson file
-                } else if (file.type == 'Feature' || file.type == 'FeatureCollection') {
-                    layer.options.mimetype = 'vector/geojson';
-                } else {
-                    throw new Error('Unsupported json file data vector');
-                }
+        const options = { buildExtent: true, crsIn: layer.projection };
 
-                if (!(layer.extent instanceof Extent)) {
-                    layer.extent = new Extent(layer.projection, layer.extent);
-                }
+        if (!(layer.extent instanceof Extent)) {
+            layer.extent = new Extent(layer.projection, layer.extent).as(parentCrs);
+        }
 
-                if (!layer.options.zoom) {
-                    layer.options.zoom = { min: 5, max: 21 };
-                }
+        if (!layer.options.zoom) {
+            layer.options.zoom = { min: 5, max: 21 };
+        }
 
-                layer.format = layer.options.mimetype;
-                layer.style = layer.style || {};
-
-                // Rasterization of data vector
-                // It shouldn't use parent's texture outside the extent
-                // Otherwise artefacts appear at the outer edge
-                layer.noTextureParentOutsideLimit = true;
-                const options = { buildExtent: true, crsIn: layer.projection };
-
-                if (layer.options.mimetype === 'vector/geojson') {
-                    layer.feature = GeoJSON2Features.parse(layer.reprojection, file, layer.extent, options);
-                    layer.extent = layer.feature.extent || layer.feature.geometry.extent;
-                } else if (layer.options.mimetype === 'vector/kml') {
-                    const geojson = togeojson.kml(file);
-                    layer.feature = GeoJSON2Features.parse(layer.reprojection, geojson, layer.extent, options);
-                    layer.extent = layer.feature.extent;
-                } else if (layer.options.mimetype === 'vector/gpx') {
-                    const geojson = togeojson.gpx(file);
+        layer.style = layer.style || {};
+
+        // Rasterization of data vector
+        // It shouldn't use parent's texture outside its extent
+        // Otherwise artefacts appear at the outer edge
+        layer.noTextureParentOutsideLimit = true;
+
+        return Fetcher.text(layer.url, layer.networkOptions).then((text) => {
+            let geojson;
+            // if it's an xml file, then it can be kml or gpx
+            if (text.startsWith('<')) {
```

Comment:
Bad assumption, for all I know there could be a `\n` (due to bad copy/paste) at the beginning of my file.